tests/std/tests/VSO_0157762_feature_test_macros/test.cpp

```diff
@@ -7,675 +7,6 @@
-#endif
-#endif
-
-#if _HAS_CXX17 && !defined(__clang__)
-// https://clang.llvm.org/cxx_status.html says P0522R0 is Partial, "Despite being
-// the resolution to a Defect Report, this feature is disabled by default in all
-// language versions, and can be enabled explicitly with the flag
-// -frelaxed-template-template-args in Clang 4 onwards. The change to the standard
-// lacks a corresponding change for template partial ordering, resulting in
-// ambiguity errors for reasonable and previously-valid code. This issue is
-// expected to be rectified soon."
-#ifndef __cpp_template_template_args
-#error __cpp_template_template_args is not defined
-#elif __cpp_template_template_args != 201611L
-#error __cpp_template_template_args is not 201611L
-#else
-STATIC_ASSERT(__cpp_template_template_args == 201611L);
-#endif
-#else
-#ifdef __cpp_template_template_args
-#error __cpp_template_template_args is defined
-#endif
-#endif
-
-// C++11, /Zc:threadSafeInit[-]
-#if defined(_M_CEE_PURE) || defined(TEST_DISABLED_THREADSAFE_STATIC_INIT)
-#ifdef __cpp_threadsafe_static_init
-#error __cpp_threadsafe_static_init is defined
-#endif
-#else
-#ifndef __cpp_threadsafe_static_init
-#error __cpp_threadsafe_static_init is not defined
-#elif __cpp_threadsafe_static_init != 200806L
-#error __cpp_threadsafe_static_init is not 200806L
-#else
-STATIC_ASSERT(__cpp_threadsafe_static_init == 200806L);
-#endif
-#endif
-
-#ifndef __cpp_unicode_characters
-#error __cpp_unicode_characters is not defined
-#elif __cpp_unicode_characters != 200704L
-#error __cpp_unicode_characters is not 200704L
-#else
-STATIC_ASSERT(__cpp_unicode_characters == 200704L);
-#endif
-
-#ifndef __cpp_unicode_literals
-#error __cpp_unicode_literals is not defined
-#elif __cpp_unicode_literals != 200710L
-#error __cpp_unicode_literals is not 200710L
-#else
-STATIC_ASSERT(__cpp_unicode_literals == 200710L);
-#endif
-
-#ifndef __cpp_user_defined_literals
-#error __cpp_user_defined_literals is not defined
-#elif __cpp_user_defined_literals != 200809L
-#error __cpp_user_defined_literals is not 200809L
-#else
-STATIC_ASSERT(__cpp_user_defined_literals == 200809L);
-#endif
-
-#ifndef __cpp_variable_templates
-#error __cpp_variable_templates is not defined
-#elif __cpp_variable_templates != 201304L
-#error __cpp_variable_templates is not 201304L
-#else
-STATIC_ASSERT(__cpp_variable_templates == 201304L);
-#endif
-
-#ifndef __cpp_variadic_templates
-#error __cpp_variadic_templates is not defined
-#elif __cpp_variadic_templates != 200704L
-#error __cpp_variadic_templates is not 200704L
-#else
-STATIC_ASSERT(__cpp_variadic_templates == 200704L);
-#endif
-
-#if _HAS_CXX17
-#ifndef __cpp_variadic_using
-#error __cpp_variadic_using is not defined
-#elif __cpp_variadic_using != 201611L
-#error __cpp_variadic_using is not 201611L
-#else
-STATIC_ASSERT(__cpp_variadic_using == 201611L);
-#endif
-#else
-#ifdef __cpp_variadic_using
-#error __cpp_variadic_using is defined
-#endif
-#endif
-
-
 // LIBRARY FEATURE-TEST MACROS

```

Comment:
Isn't this comment now redundant?

Reply:
Yeah, good catch - I noticed that too, but figured it wasn't really harmful. I'm ok with saying "yeah, it's redundant, but it's not worth delaying these PRs further". If the comment were actively misleading now, I think that would meet a threshold for changing it.

Reply:
OTOH it might be a good idea to change the comment to say that the core language feature test macro tests are part of the compiler test suite or alternatively changing the directory name to say `library_feature_test_macros`?

Reply:
I'm not disturbed by the state of this test. If someone else is, feel free to submit a followup PR.

Reply:
I'd approve a PR changing the directory name (but like Casey, I feel no need to make such a PR myself). Changing the comment to talk about the compiler test suite would be strange because that test suite isn't on GitHub.

Reply:
Yeah I don't feel that strongly about it to make a PR. Mainly just thinking out loud.